oarepo_ui/templates/components/datafields/files/FilesViewer.jinja

```diff
@@ -1,30 +1,47 @@
-{# def files #}
+{# def files #} {% if files.entries|length > 0 %}
 
-{%if files.entries|length > 0 %}
 <table class="ui table striped rel-mb-1">
-    <thead>
-        <tr>
-            <th colspan="3"><h2 class="ui header">{{ _("Files") }}</h2></th>
-        </tr>
-        <tr>
-            <th>{{_("File")}}</th>
-            <th>{{_("Size")}}</th>
-            <th>{{_("Actions")}}</th>
-        </tr>
-    </thead>
-    <tbody>
-        {% for file in array(files.entries) %}
-            <tr>
-                <td>
-                    <a href="{{ file.links.content }}">{{ file.key }}</a>
-                </td>
-                <td>{{ file.size|filesizeformat }}</td>
-                <td>
-                    <i class="ui icon zoom"></i>
-                    <a href="{{ file.links.content }}"><i class="ui icon download"></i></a>
-                </td>
-            </tr>
-        {% endfor %}
-    </tbody>
+  <thead>
+    <tr>
+      <th colspan="3">
+        <h2 class="ui header">{{ _("Files") }}</h2>
+      </th>
+    </tr>
+    <tr>
+      <th>{{ _("File") }}</th>
+      <th>{{ _("Size") }}</th>
+      <th>{{ _("Actions") }}</th>
+    </tr>
+  </thead>
+  <tbody>
+    {% for file in array(files.entries) %}
+    <tr>
+      <td>
+        <a href="{{ file.links.content }}">{{ file.key }}</a>
+      </td>
+      <td>{{ file.size | filesizeformat }}</td>
+      <td>
+        {% if file.links.preview  %}
+        <button class="ui button">
+          <i
+            class="ui icon zoom openPreviewIcon"
+            data-preview-link="{{ file.links.preview }}"
+          ></i>
+        </button>
+        {% endif %}
+        <button class="ui button">
+          <a href="{{ file.links.content }}"
+            ><i class="ui icon download"></i
+          ></a>
+        </button>
+      </td>
+    </tr>
+    {% endfor %}
+  </tbody>
 </table>
+
+<div class="ui modal fullscreen" id="preview-modal">
+  <i class="close icon"></i>
```

Comment:
Same semantic problem as with preview button, this has to be a button, not just some icon